### PR TITLE
Use non-zero resync period as last insurance in local volume provisioner

### PR DESCRIPTION
**Pull request by cofyc**
Fixes https://github.com/kubernetes-incubator/external-storage/issues/795.

- Add `MinResyncPeriod` configuration, defaults: 5m0s.
- Update JobController to use default resync period.
- Test `TestLoadProvisionerConfigs ` with real cases.
- Add example for customizing resync period.
- Update test/run.sh to fail on errors.

### CODE REVIEW

File: local-volume/provisioner/pkg/controller/controller.go [hunk 1]
```diff
@@ -49,6 +50,11 @@ func StartLocalController(client *kubernetes.Clientset, ptable deleter.ProcTable
 	broadcaster.StartRecordingToSink(&v1core.EventSinkImpl{Interface: v1core.New(client.CoreV1().RESTClient()).Events("")})
 	recorder := broadcaster.NewRecorder(scheme.Scheme, v1.EventSource{Component: provisionerName})
 
+	// We choose a random resync period between MinResyncPeriod and 2 *
+	// MinResyncPeriod, so that local provisioners deployed on multiple nodes
+	// at same time don't list the apiserver simultaneously.
+	resyncPeriod := time.Duration(float64(config.MinResyncPeriod.Nanoseconds()) * (1 + rand.Float64()))
```
Comment (msau42) [inline local-volume/provisioner/pkg/controller/controller.go]: The final value should be multiplied by time.Nanosecond.  Also, do we really need Nanosecond granularlity? I think probably Seconds should be sufficient.

Comment (cofyc) [reply local-volume/provisioner/pkg/controller/controller.go]: Fixed in https://github.com/kubernetes-incubator/external-storage/pull/800/commits/fec4ef952a9006121e3b5ac61b3149cb2eb46b38. Also initialized default source in main.go.

File: local-volume/provisioner/pkg/controller/controller.go [hunk 2]
```diff
@@ -53,7 +53,7 @@ func StartLocalController(client *kubernetes.Clientset, ptable deleter.ProcTable
 	// We choose a random resync period between MinResyncPeriod and 2 *
 	// MinResyncPeriod, so that local provisioners deployed on multiple nodes
 	// at same time don't list the apiserver simultaneously.
-	resyncPeriod := time.Duration(float64(config.MinResyncPeriod.Nanoseconds()) * (1 + rand.Float64()))
+	resyncPeriod := time.Duration(config.MinResyncPeriod.Seconds()*(1+rand.Float64())) * time.Second
```
Comment (msau42) [inline local-volume/provisioner/pkg/controller/controller.go]: I think if you don't cast it to float64 like before, then you don't need to multiple by time.Second at the end.

File: local-volume/provisioner/cmd/main.go [hunk 3]
```diff
@@ -43,6 +44,7 @@ var (
 )
 
 func main() {
+	rand.Seed(time.Now().UTC().UnixNano())
```
Comment (msau42) [inline local-volume/provisioner/cmd/main.go]: This may be causing mac osx build to fail?

Comment (cofyc) [reply local-volume/provisioner/cmd/main.go]: helm install script, fixed in https://github.com/kubernetes-incubator/external-storage/pull/806.

File: local-volume/provisioner/pkg/controller/controller.go (hunk 2 above)
Comment (cofyc) [reply local-volume/provisioner/pkg/controller/controller.go]: `Seconds()` method divides it by time.Second to get number of seconds, so we need to multiple by time.Second at the end. If you prefer not to multiple by time.Second, we need to calculate it using nanoseconds. Actually time.Duration is int64 in nanoseconds.

Demo: https://play.golang.org/p/taL1ywwQ1wt

**Comment by cofyc** [comment]
Rebased due to conflicts with https://github.com/kubernetes-incubator/external-storage/pull/799.

**Pull request merged**